removed jenkinsfile and added gitlab ci yaml

leahnp opened the pull request:
Who ever reviews this, please double check the publishing of master:latest and master:tagged fits our best practices!

Fixes: https://github.com/samsung-cnct/kraken-tools/issues/157

coffeepac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,210 @@
+  variables:
+    DOCKER_REGISTRY: $CI_REGISTRY
+    DOCKER_PROJECT: $CI_PROJECT_PATH
+    DOCKER_TAG: branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID
```
is this going to work for master images tagged `:latest`?  I don't really know how the cleanup image works.

leahnp replied:
I was under the impression no one would ever explicit set `CI_COMMIT_TAG` to `latest` - but each CI run would auto `publish` master with a `latest` tag. So a tag that is explicitly set would be something like `v0.0.0`. With this workflow the gitlab image registry would never have an image with a `latest` tag in it.

Is this assumption incorrect?

guineveresaenger replied:
I believe your assumption is correct. `:latest` only goes to Quay; on gitlab the tag would be what is set in the build stage.

coffeepac replied:
my apologies, I am maybe commenting out of turn and asking questions, but I am a little confused as to where the master branch will be built (just noticing this now and this is probably the cause of my confusion).  I *think* that because there is both a `-branches` and a `-master` filter tag for some stages that `-branches` doesn't cover the master branch.  so:
a. where does the image get built for the master branch?
b. regarding `:latest` it appears that the master publish pulls a `:latest` tag from the gitlab docker repo and then that image gets pushed (is that some type of magic where `:latest` is always simply the last image built by this job across any branch?)

leahnp commented:
@coffeepac made a few more commits and responded to your question (with another question)

coffeepac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,207 @@
+    IMAGE_URI: $CI_REGISTRY_IMAGE:branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID
+  only:
+    - branches
+    - master
```
I *think* `-master` is unnecessary here.

coffeepac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,207 @@
+    IMAGE_URI: $CI_REGISTRY_IMAGE:branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID
+  only:
+    - branches
+    - master
```
I *think* `-master` is unnecessary here

coffeepac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,207 @@
+publish-latest-master:
+  stage: publish
+  variables:
+    IMAGE_URI: $CI_REGISTRY_IMAGE:latest
```
I think this should not be `:latest` but should be `:branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID` because that's what tag gets created in the `build-branch` stage.

coffeepac commented on .gitlab-ci.yml:
```diff
@@ -0,0 +1,207 @@
+    DOCKER_TAG: branch-$CI_COMMIT_REF_NAME-$CI_PIPELINE_ID
+  only:
+    - branches
+    - master
```
maybe don't need `-master` here

leahnp commented:
@coffeepac I think I have addressed your concerns - I was slowly understanding the branch/master relationship as well on Friday afternoon. I need to double check the Quay env vars and then this should be good to go.

leahnp commented:
Publish is pushing to quay! This is ready for final review + merge.

guineveresaenger commented:
This isn't really a block to merging this PR, but, we should probably remove the GitHub Jenkins integration for this.

coffeepac commented on tests/aws-testing.sh:
```diff
@@ -12,7 +12,7 @@ echo "prep complete, create config"
 
 echo "modify config in-place"
-build-scripts/update-generated-config.sh cluster/aws/config.yaml ${JOB_BASE_NAME}-${BUILD_ID}
+build-scripts/update-generated-config.sh cluster/aws/config.yaml ci-${BUILD_ID}
```
can we replace this with 'kt' or 'ktool'?  this job will be operating in the same namespace as `kraken` and `kraken-lib` builds for cluster names so we should be adding something to ensure we don't get name collisions.

however, I'm not sure how many characters we have to work with, so `kt` would be fine but `kraken-tools` would be best, but that may cause ELB names to be too long.

leahnp replied:
`krakentools-` keeps us under 22 chars - updated.

coffeepac commented on tests/gke-testing.sh:
```diff
@@ -14,7 +14,7 @@ mkdir -p cluster/gke
 
 echo "modify config in-place"
-build-scripts/update-generated-config.sh cluster/gke/config.yaml ${JOB_BASE_NAME}-${BUILD_ID}
+build-scripts/update-generated-config.sh cluster/gke/config.yaml ci-${BUILD_ID}
```
same as above, needs to be specific to `kraken-tools` in some way.

leahnp commented:
@guineveresaenger removed webhooks for Jenkins - thanks for the reminder

guineveresaenger commented:
Huh weird. Jenkins is still there. Will merge for now
